<a href="https://colab.research.google.com/github/tal21-linares/curso-ia-para-economia/blob/main/Taller_Reg_Pol_AlejandroVelandia_TaliaLinares.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<a href="https://colab.research.google.com/github/LinaMariaCastro/curso-ia-para-economia/blob/main/clases/5_Aprendizaje_supervisado/2_Taller_Regresion_Polinomica.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Forma de entrega:**

- Nombrar el archivo de la siguiente forma: “Taller_Reg_Pol_apellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.cloud.microsoft/r/AUmhqMjhUK.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

**Forma de entrega:**

- Nombrar el archivo de la siguiente forma: “Taller_Reg_Pol_apellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.cloud.microsoft/r/AUmhqMjhUK.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

# **Inteligencia Artificial con Aplicaciones en Economía I**

- 👩‍🏫 **Profesora:** [Lina María Castro](https://www.linkedin.com/in/lina-maria-castro)  
- 📧 **Email:** [lmcastroco@gmail.com](mailto:lmcastroco@gmail.com)  
- 🎓 **Universidad:** Universidad Externado de Colombia - Facultad de Economía

# **Taller: Regresión Polinómica, Subajuste, Sobreajuste y Regularización**

**IMPORTANTE**: Guarda una copia de este notebook en tu Google Drive o computador.

**Taller en grupos de 3**

**Nombres estudiantes:**



*   Alejandro Velandia
*   Talia Linares


**Forma de entrega:**

- Nombrar el archivo de la siguiente forma: “Taller_Reg_Pol_apellidos.ipynb”.
- Suba el Jupyter Notebook a su cuenta en Github y envíe el link en el siguiente Forms: https://forms.cloud.microsoft/r/AUmhqMjhUK.

**IMPORTANTE:** No se recibirán talleres en Google Colab, el notebook debe estar subido en Github.

**Plazo de entrega:**

30 de abril de 2026, máximo a las 11:59 p.m. Tenga en cuenta que luego de esa hora el formulario en forms se cierra. El Jupupyter Notebook también debe quedar subido en Github antes de esa hora.

**Instrucciones Generales:**

Completa el código en las celdas marcadas con `### TU CÓDIGO AQUÍ ###`. Puedes añadir más celdas si lo requieres.

## **Situación**

Una importante firma de inversión inmobiliaria te ha contratado como consultor de ciencia de datos. Su proceso actual de valoración de propiedades es lento y se basa en la intuición de unos pocos expertos. Quieren que desarrolles un modelo de machine learning para predecir el precio de venta de las viviendas (`SalePrice`) de forma más precisa y sistemática.

Te han entregado el dataset "Ames Housing", que contiene una gran cantidad de información sobre viviendas vendidas recientemente. Tu tarea es construir el mejor modelo posible, pero más importante aún, justificar por qué tu modelo es robusto y fiable, explicando cómo has manejado la complejidad y el riesgo de sobreajuste.

### **Ejercicio 1: Carga y Preparación Inicial de los Datos**

Primero, carguemos las librerías necesarias y el dataset. Nos enfocaremos en un subconjunto de las variables numéricas para mantener el taller manejable, pero el principio se aplica a todo el dataset.

In [ ]:
# Importación de librerías estándar
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

# Importación de herramientas de scikit-learn
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression, Ridge, Lasso
from sklearn.preprocessing import PolynomialFeatures, StandardScaler
from sklearn.pipeline import Pipeline
from sklearn.metrics import mean_squared_error

Mejorar visualización de dataframes y gráficos

In [ ]:
# Que muestre todas las columnas
pd.options.display.max_columns = None
# En los dataframes, mostrar los float con dos decimales
pd.options.display.float_format = '{:,.2f}'.format

# Configuraciones para una mejor visualización
sns.set(style='whitegrid')
plt.rcParams['figure.figsize'] = (10, 6)

Cargar el dataset

In [ ]:
# Usamos el dataset Ames Housing desde su fuente original (requiere internet)
url = 'http://jse.amstat.org/v19n3/decock/AmesHousing.txt'
df = pd.read_csv(url, sep='\t')
df.head()

,Order,PID,MS SubClass,MS Zoning,Lot Frontage,Lot Area,Street,Alley,Lot Shape,Land Contour,Utilities,Lot Config,Land Slope,Neighborhood,Condition 1,Condition 2,Bldg Type,House Style,Overall Qual,Overall Cond,Year Built,Year Remod/Add,Roof Style,Roof Matl,Exterior 1st,Exterior 2nd,Mas Vnr Type,Mas Vnr Area,Exter Qual,Exter Cond,Foundation,Bsmt Qual,Bsmt Cond,Bsmt Exposure,BsmtFin Type 1,BsmtFin SF 1,BsmtFin Type 2,BsmtFin SF 2,Bsmt Unf SF,Total Bsmt SF,Heating,Heating QC,Central Air,Electrical,1st Flr SF,2nd Flr SF,Low Qual Fin SF,Gr Liv Area,Bsmt Full Bath,Bsmt Half Bath,Full Bath,Half Bath,Bedroom AbvGr,Kitchen AbvGr,Kitchen Qual,TotRms AbvGrd,Functional,Fireplaces,Fireplace Qu,Garage Type,Garage Yr Blt,Garage Finish,Garage Cars,Garage Area,Garage Qual,Garage Cond,Paved Drive,Wood Deck SF,Open Porch SF,Enclosed Porch,3Ssn Porch,Screen Porch,Pool Area,Pool QC,Fence,Misc Feature,Misc Val,Mo Sold,Yr Sold,Sale Type,Sale Condition,SalePrice
0,1,526301100,20,RL,141.00,31770,Pave,NaN,IR1,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,6,5,1960,1960,Hip,CompShg,BrkFace,Plywood,Stone,112.00,TA,TA,CBlock,TA,Gd,Gd,BLQ,639.00,Unf,0.00,441.00,"1,080.00",GasA,Fa,Y,SBrkr,1656,0,0,1656,1.00,0.00,1,0,3,1,TA,7,Typ,2,Gd,Attchd,"1,960.00",Fin,2.00,528.00,TA,TA,P,210,62,0,0,0,0,NaN,NaN,NaN,0,5,2010,WD,Normal,215000
1,2,526350040,20,RH,80.00,11622,Pave,NaN,Reg,Lvl,AllPub,Inside,Gtl,NAmes,Feedr,Norm,1Fam,1Story,5,6,1961,1961,Gable,CompShg,VinylSd,VinylSd,NaN,0.00,TA,TA,CBlock,TA,TA,No,Rec,468.00,LwQ,144.00,270.00,882.00,GasA,TA,Y,SBrkr,896,0,0,896,0.00,0.00,1,0,2,1,TA,5,Typ,0,NaN,Attchd,"1,961.00",Unf,1.00,730.00,TA,TA,Y,140,0,0,0,120,0,NaN,MnPrv,NaN,0,6,2010,WD,Normal,105000
2,3,526351010,20,RL,81.00,14267,Pave,NaN,IR1,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,6,6,1958,1958,Hip,CompShg,Wd Sdng,Wd Sdng,BrkFace,108.00,TA,TA,CBlock,TA,TA,No,ALQ,923.00,Unf,0.00,406.00,"1,329.00",GasA,TA,Y,SBrkr,1329,0,0,1329,0.00,0.00,1,1,3,1,Gd,6,Typ,0,NaN,Attchd,"1,958.00",Unf,1.00,312.00,TA,TA,Y,393,36,0,0,0,0,NaN,NaN,Gar2,12500,6,2010,WD,Normal,172000
3,4,526353030,20,RL,93.00,11160,Pave,NaN,Reg,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,7,5,1968,1968,Hip,CompShg,BrkFace,BrkFace,NaN,0.00,Gd,TA,CBlock,TA,TA,No,ALQ,"1,065.00",Unf,0.00,"1,045.00","2,110.00",GasA,Ex,Y,SBrkr,2110,0,0,2110,1.00,0.00,2,1,3,1,Ex,8,Typ,2,TA,Attchd,"1,968.00",Fin,2.00,522.00,TA,TA,Y,0,0,0,0,0,0,NaN,NaN,NaN,0,4,2010,WD,Normal,244000
4,5,527105010,60,RL,74.00,13830,Pave,NaN,IR1,Lvl,AllPub,Inside,Gtl,Gilbert,Norm,Norm,1Fam,2Story,5,5,1997,1998,Gable,CompShg,VinylSd,VinylSd,NaN,0.00,TA,TA,PConc,Gd,TA,No,GLQ,791.00,Unf,0.00,137.00,928.00,GasA,Gd,Y,SBrkr,928,701,0,1629,0.00,0.00,2,1,3,1,TA,6,Typ,1,TA,Attchd,"1,997.00",Fin,2.00,482.00,TA,TA,Y,212,34,0,0,0,0,NaN,MnPrv,NaN,0,3,2010,WD,Normal,189900


In [ ]:
# Para este taller, solo usaremos algunas columnas clave para simplificar.
df_sub = df[['Overall Qual', 'Gr Liv Area', 'SalePrice']].copy()
df_sub.columns = ['OverallQual', 'GrLivArea', 'SalePrice'] # Renombrar para facilidad
df_sub.head()

,OverallQual,GrLivArea,SalePrice
0,6,1656,215000
1,5,896,105000
2,6,1329,172000
3,7,2110,244000
4,5,1629,189900


**Explicación de las variables del dataset reducido**

1. SalePrice: Precio de Venta.

Esta es la variable objetivo. Es el precio final por el cual se vendió la propiedad, medido en dólares estadounidenses.

2. OverallQual: Calidad General.

Es una variable ordinal que califica la calidad general del material y el acabado de la casa. Es una de las variables predictoras (X) más importantes.

Escala: Va de 1 a 10.

10: Muy Excelente

9: Excelente

...

2: Pobre

1: Muy Pobre

3. GrLivArea: Área Habitable sobre el Nivel del Suelo.

Es una variable numérica que mide el total de metros cuadrados de área habitable que está por encima del nivel del suelo. No incluye el área del sótano. Es una de las variables predictoras (X) más fuertes, ya que, lógicamente, casas más grandes tienden a ser más caras.

In [ ]:
df_sub.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2930 entries, 0 to 2929
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   OverallQual  2930 non-null   int64
 1   GrLivArea    2930 non-null   int64
 2   SalePrice    2930 non-null   int64
dtypes: int64(3)
memory usage: 68.8 KB


### **Ejercicio 2: Dividir el conjunto de datos**

El método `.info()` muestra que no hay nulos en nuestro subconjunto. ¡Perfecto! Ahora, definamos nuestras variables `X` (predictoras) e `y` (objetivo) y dividamos los datos.

In [ ]:
X = df_sub[['OverallQual', 'GrLivArea']]
y = df_sub['SalePrice']


In [ ]:
# Dividir en entrenamiento y prueba (70/30)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.3, random_state=42
)


### **Ejercicio 3: Modelo con Sobreajuste**

Vamos a crear un modelo muy complejo para ver qué tan mal puede generalizar.

**Crear un Modelo Polinómico de Grado 5**

Usa `Pipeline` para combinar `PolynomialFeatures` (grado 5), `StandardScaler` y `LinearRegression`. Entrénalo con los datos de entrenamiento.

In [ ]:
# Crear el pipeline para el modelo polinómico
modelo_polinomico = Pipeline([
    ('poly', PolynomialFeatures(degree=5)),
    ('scaler', StandardScaler()),
    ('model', LinearRegression())
])


In [ ]:
# Entrenar el modelo
modelo_polinomico.fit(X_train, y_train)


Pipeline(steps=[('poly', PolynomialFeatures(degree=5)),
                ('scaler', StandardScaler()), ('model', LinearRegression())])

In [ ]:
# Calcular el error (RMSE) en entrenamiento y prueba y realizar un print de estos
#Predicciones
y_train_pred_poly = modelo_polinomico.predict(X_train)
y_test_pred_poly = modelo_polinomico.predict(X_test)

# Cálculo del error (RMSE)
rmse_train_poly = np.sqrt(mean_squared_error(y_train, y_train_pred_poly))
rmse_test_poly = np.sqrt(mean_squared_error(y_test, y_test_pred_poly))

print(f"RMSE en Entrenamiento (Polinómico): ${rmse_train_poly:,.2f}")
print(f"RMSE en Prueba (Polinómico): ${rmse_test_poly:,.2f}")


RMSE en Entrenamiento (Polinómico): $34,152.35
RMSE en Prueba (Polinómico): $41,101.60


**Pregunta:** ¿Qué indica la diferencia que observas entre el error de entrenamiento y el de prueba? ¿Le recomendarías este modelo a la firma inmobiliaria? ¿Por qué?

La diferencia significativa entre el error de entrenamiento y el de prueba indica la presencia de sobreajuste. El modelo ha aprendido en exceso los patrones específicos del conjunto de entrenamiento, incluyendo el ruido, lo que reduce su capacidad de generalización.

Por esta razón, no se recomienda su implementación en un entorno real, ya que su desempeño fuera de muestra es inestable y poco fiable para la toma de decisiones.

### **Ejercicio 4: Aplicar Regularización**

Ahora, vamos a "curar" el sobreajuste. Usaremos los mismos `PolynomialFeatures` de grado 5, pero cambiaremos el modelo de regresión.

**Implementar Regresión Ridge**

Copia el pipeline anterior, pero reemplaza `LinearRegression` con `Ridge(alpha=10)`. `alpha` es la fuerza de la penalización.

In [ ]:
# Crear el pipeline para Ridge
modelo_ridge = Pipeline([
    ('poly', PolynomialFeatures(degree=5)),
    ('scaler', StandardScaler()),
    ('model', Ridge(alpha=10))
])


In [ ]:
# Entrenar el modelo
modelo_ridge.fit(X_train, y_train)


Pipeline(steps=[('poly', PolynomialFeatures(degree=5)),
                ('scaler', StandardScaler()), ('model', Ridge(alpha=10))])

In [ ]:
# Calcular el error (RMSE) en entrenamiento y prueba y realizar un print de estos
# Predicciones
y_train_pred_ridge = modelo_ridge.predict(X_train)
y_test_pred_ridge = modelo_ridge.predict(X_test)

# Cálculo del error (RMSE)
rmse_train_ridge = np.sqrt(mean_squared_error(y_train, y_train_pred_ridge))
rmse_test_ridge = np.sqrt(mean_squared_error(y_test, y_test_pred_ridge))

print(f"RMSE en Entrenamiento (Ridge): ${rmse_train_ridge:,.2f}")
print(f"RMSE en Prueba (Ridge): ${rmse_test_ridge:,.2f}")


RMSE en Entrenamiento (Ridge): $35,204.89
RMSE en Prueba (Ridge): $35,206.63


**Interpreta los resultados:**
El modelo Ridge presenta un RMSE prácticamente idéntico en entrenamiento `($35,204.89) y prueba ($35,206.63)`, lo que indica una excelente capacidad de generalización y ausencia de sobreajuste. La regularización L2 logra controlar eficazmente la complejidad del modelo polinómico, estabilizando los coeficientes y permitiendo que el modelo mantenga un desempeño consistente en datos no observados, lo que lo convierte en una opción robusta y confiable para la predicción de precios de vivienda.


**Implementar Regresión Lasso**

Haz lo mismo, pero ahora con `Lasso(alpha=500, max_iter=10000)`. Lasso necesita un `alpha` más grande (porque la penalización L1 es diferente) y a veces más `max_iter` para converger.

In [ ]:
# Crear el pipeline para Lasso
modelo_lasso = Pipeline([
    ('poly', PolynomialFeatures(degree=5)),
    ('scaler', StandardScaler()),
    ('model', Lasso(alpha=500, max_iter=10000))
])

In [ ]:
# Entrenar el modelo
modelo_lasso.fit(X_train, y_train)

Pipeline(steps=[('poly', PolynomialFeatures(degree=5)),
                ('scaler', StandardScaler()),
                ('model', Lasso(alpha=500, max_iter=10000))])

In [ ]:
# Calcular el error (RMSE) en entrenamiento y prueba y realizar un print de estos
# Predicciones
y_train_pred_lasso = modelo_lasso.predict(X_train)
y_test_pred_lasso = modelo_lasso.predict(X_test)

# Cálculo del error (RMSE)
rmse_train_lasso = np.sqrt(mean_squared_error(y_train, y_train_pred_lasso))
rmse_test_lasso = np.sqrt(mean_squared_error(y_test, y_test_pred_lasso))

print(f"RMSE en Entrenamiento (Lasso): ${rmse_train_lasso:,.2f}")
print(f"RMSE en Prueba (Lasso): ${rmse_test_lasso:,.2f}")


RMSE en Entrenamiento (Lasso): $35,352.92
RMSE en Prueba (Lasso): $35,447.71


**Interpreta los resultados:**

El modelo Lasso muestra un RMSE de` $35,352.92 en entrenamiento y $35,447.71 en prueba`, evidenciando también una buena capacidad de generalización, aunque con un error ligeramente mayor que Ridge. Esto sugiere que, si bien la regularización L1 ayuda a reducir el sobreajuste y simplificar el modelo mediante la selección de variables, en este caso puede estar eliminando información relevante, lo que resulta en una leve pérdida de precisión en las predicciones.

**Selección de Variables con Lasso**

Una de las grandes ventajas de Lasso es que puede eliminar variables. Vamos a ver cuántas de las características polinómicas que creamos (ej. `OverallQual^2`, `GrLivArea^5`, `OverallQual * GrLivArea^4`, etc.) fueron eliminadas.

In [ ]:
# Extraer los coeficientes del modelo Lasso
coefs_lasso = modelo_lasso.named_steps['model'].coef_


In [ ]:
# Extraer los nombres de las características generadas
# Guíate por el siguiente código: feature_names = pipeline_lasso.named_steps['poly_features'].get_feature_names_out(X.columns)
feature_names = modelo_lasso.named_steps['poly'].get_feature_names_out(X.columns)

In [ ]:
# Contar cuántos coeficientes son exactamente cero
num_zero_coeffs = np.sum(coefs_lasso == 0)


In [ ]:
# Realizar un print de:
# Número total de características polinómicas generadas
# Número de características eliminadas por Lasso (coef = 0)
# Número de características CONSERVADAS
# Porcentaje de variables eliminadas
# Lista con los nombres de las características conservadas por Lasso (coef != 0)

# Número total de características
num_total = len(coefs_lasso)

# Número de características conservadas
num_nonzero = num_total - num_zero_coeffs

# Porcentaje eliminadas
percent_zero = (num_zero_coeffs / num_total) * 100

# Características conservadas
selected_features = feature_names[coefs_lasso != 0]

print(f"Número total de características polinómicas: {num_total}")
print(f"Número de características eliminadas (coef = 0): {num_zero_coeffs}")
print(f"Número de características CONSERVADAS: {num_nonzero}")
print(f"Porcentaje de variables eliminadas: {percent_zero:.2f}%")

print("\nCaracterísticas conservadas por Lasso:")
print(selected_features)


Número total de características polinómicas: 21
Número de características eliminadas (coef = 0): 15
Número de características CONSERVADAS: 6
Porcentaje de variables eliminadas: 71.43%

Características conservadas por Lasso:
['OverallQual' 'OverallQual GrLivArea' 'OverallQual^2 GrLivArea'
 'OverallQual^3 GrLivArea' 'OverallQual^5' 'GrLivArea^5']


**Interpreta los resultados:**
El modelo Lasso eliminó 15 de las 21 características polinómicas (71.43%), conservando únicamente 6 variables, lo que evidencia una fuerte reducción de la complejidad del modelo. Esto indica que muchas de las combinaciones polinómicas generadas no aportaban valor significativo a la predicción del precio y, por tanto, fueron descartadas automáticamente.

Las variables que se conservaron muestran un patrón claro: la mayoría están relacionadas con OverallQual (calidad de la vivienda), ya sea de forma directa o en interacción con el área habitable (GrLivArea). Esto sugiere que la calidad es un factor estructural clave en la determinación del precio, mientras que el área influye principalmente en combinación con esta. Además, la presencia de términos como GrLivArea^5 indica que el tamaño también tiene un efecto no lineal relevante, aunque más específico.

En conjunto, estos resultados reflejan que Lasso no solo mejora la generalización del modelo, sino que también actúa como un mecanismo de selección de variables, permitiendo identificar de manera automática cuáles son los factores más importantes. Esto hace que el modelo sea más simple, interpretable y enfocado en las relaciones realmente significativas dentro del mercado inmobiliario.

### **Ejercicio 5: Conclusión y Recomendación para el Cliente**

**Resumir los Resultados**

Crea un `DataFrame` de pandas que compare el RMSE de entrenamiento y prueba de los tres modelos (Polinómico, Ridge, Lasso) y ordena los modelos según el RMSE de prueba de menor a mayor.

In [ ]:
# Crear DataFrame de resultados
# Crear DataFrame con resultados
results = pd.DataFrame({
    'Modelo': [
        'Polinómico Grado 5 (Sobreajuste)',
        'Ridge (Regularizado L2)',
        'Lasso (Regularizado L1)'
    ],
    'RMSE en Entrenamiento': [
        rmse_train_poly,
        rmse_train_ridge,
        rmse_train_lasso
    ],
    'RMSE en Prueba': [
        rmse_test_poly,
        rmse_test_ridge,
        rmse_test_lasso
    ]
})

# Ordenar por mejor desempeño en prueba
results_sorted = results.sort_values(by='RMSE en Prueba')

print("Tabla Comparativa de Rendimiento (Error en Dólares)")

display(results_sorted.style.format({
    'RMSE en Entrenamiento': '${:,.2f}',
    'RMSE en Prueba': '${:,.2f}'
}).background_gradient(cmap='Reds', subset=['RMSE en Prueba']))


Tabla Comparativa de Rendimiento (Error en Dólares)


,Modelo,RMSE en Entrenamiento,RMSE en Prueba
1,Ridge (Regularizado L2),"$35,204.89","$35,206.63"
2,Lasso (Regularizado L1),"$35,352.92","$35,447.71"
0,Polinómico Grado 5 (Sobreajuste),"$34,152.35","$41,101.60"


**Pregunta Final:**

Basado en tu análisis, ¿qué modelo le recomendarías a la firma inmobiliaria? Tu respuesta debe incluir:

1.  Una recomendación clara del mejor modelo.
2.  Una explicación en términos sencillos (para un gerente, no para un científico de datos) de por qué el modelo polinómico simple no era una buena idea, usando el concepto de "memorizar vs. generalizar".
3.  Una descripción de la ventaja principal del modelo Lasso en cuanto a la interpretabilidad y la selección de las características más importantes.

Recomendaría el uso del modelo Ridge (regularización L2), ya que presenta el mejor desempeño en términos de precisión y estabilidad. Sus resultados muestran un error prácticamente igual entre entrenamiento y prueba, lo que indica que es un modelo confiable y capaz de predecir correctamente el precio de nuevas propiedades, no solo de aquellas que ya ha visto.

El modelo polinómico sin regularización no es una buena opción porque, aunque parecía muy preciso en los datos de entrenamiento, en realidad estaba “memorizando” los datos en lugar de aprender patrones generales. Es decir, se ajustaba demasiado a los detalles específicos (incluso al ruido), lo que hacía que su desempeño fuera pobre al enfrentarse a nuevas viviendas. Para un gerente, esto sería como un empleado que aprende respuestas exactas de memoria, pero no sabe cómo actuar ante situaciones nuevas.

Por otro lado, el modelo Lasso ofrece una ventaja importante en términos de interpretabilidad. A diferencia de otros modelos, Lasso puede eliminar automáticamente variables poco relevantes al llevar sus coeficientes a cero, lo que permite identificar cuáles características realmente influyen en el precio de una vivienda. Esto no solo simplifica el modelo, sino que también facilita la toma de decisiones estratégicas, al resaltar los factores más importantes en la valoración inmobiliaria.